Lib/idlelib/configdialog.py

```diff
@@ -2122,6 +2125,342 @@ def update_help_changes(self):
+
+    def set_edit_delete_state(self, event=None):
+        """Toggle the state for the add & delete buttons based on list entries."""
+        enable = (
+            self.checkerlist.size() >= 0 and
+            self.checkerlist.curselection()
+        )
+        new_state = ('!disabled' if enable else 'disabled',)
+        self.button_checkerlist_edit.state(new_state)
+        self.button_checkerlist_remove.state(new_state)
+
+    def load_checkers_cfg(self):
+        """Fill self.checkers with data from the default and user configs."""
+        checker_names = sorted(
+            set(idleConf.GetSectionList('default', 'checkers')) |
+            set(idleConf.GetSectionList('user', 'checkers'))
+        )
+        self.checkerlist.delete(0, END)
+        for checker_name in checker_names:
+            self.checkerlist.insert(END, checker_name)
+
+    def get_checker_names(self):
+        return self.checkerlist.get(0, END)
+
+
+class ConfigCheckerDialog(Toplevel):
+
+    def __init__(self, page, callback, checker_name=None):
+        super().__init__(page)
+        self.existing_checker_names = page.get_checker_names()
+        self.callback = callback
+
+        if checker_name is None:
+            checker_data = {
+                name: (default or '') if type_ == 'str' else default
+                for name, type_, default, warn_on_default
+                in CHECKER_PARAMS
+            }
+        else:
+            checker_data = get_checker_config(checker_name)
+        self.orig_checker_data = checker_data
+
+        self.vars = {}
+        for name, type_, *_ in CHECKER_PARAMS:
+            var_cls = {'bool': BooleanVar, 'str': StringVar}[type_]
+            self.vars[name] = var = var_cls(self)
+            var.set(checker_data[name])
+
+        self.vars['call_string'] = StringVar(self)
+        self.update_call_string()
+        self.vars['command'].trace('w', self.update_call_string)
+        self.vars['additional'].trace('w', self.update_call_string)
+
+        self.grab_set()
+        self.resizable(width=False, height=False)
+        self.create_widgets()
+
+    def create_widgets(self):
+        parent = self.master
+        self.configure(borderwidth=5)
+        if self.orig_checker_data['name']:
+            title = 'Edit {} checker'.format(self.orig_checker_data['name'])
+        else:
+            title = 'Config new checker'
+        self.wm_title(title)
+        self.geometry('+%d+%d' % (parent.winfo_rootx() + 30,
+                                  parent.winfo_rooty() + 30))
+        self.transient(parent)
+        self.focus_set()
+        # frames creation
+        optionsFrame = Frame(self)
+        buttonFrame = Frame(self)
+
+        # optionsFrame
+        nameLabel = Label(optionsFrame, text='Name of Checker')
+        commandLabel = Label(optionsFrame, text='Command')
+        additionalLabel = Label(optionsFrame, text='Additional Args')
+        currentCallStringLabel = Label(optionsFrame,
```

Comment:
The recent changes to `configdialog` have taken out the use of `Label` (and most other widget types) in the variable name.  Some of them like `list` and `frame` are still used because it clarifies the usage.